apps/wallet/src/background/keyring/Vault.ts

```diff
@@ -0,0 +1,64 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+import { encrypt, decrypt } from '_shared/cryptography/keystore';
+import {
+    entropyToMnemonic,
+    entropyToSerialized,
+    mnemonicToEntropy,
+    toEntropy,
+    validateEntropy,
+} from '_shared/utils/bip39';
+
+export const LATEST_VAULT_VERSION = 1;
+
+export type StoredData = string | { v: number; data: string };
+
+/**
+ * Holds the mnemonic of the wallet and provides functionality to create/encrypt/decrypt it.
+ */
+export class Vault {
+    public readonly entropy: Uint8Array;
+
+    public static async from(
+        password: string,
+        data: StoredData,
+        onMigrateCallback?: (vault: Vault) => Promise<void>
+    ) {
+        let entropy: Uint8Array | null = null;
+        let doMigrate = false;
+        if (typeof data === 'string') {
+            entropy = mnemonicToEntropy(await decrypt<string>(password, data));
```

Comment:
Need to fix this - currently mnemonic is not plain text but object (the buffer in json)

Reply:
Done